### PR TITLE
Migrate ConditionForwarding to OSSA

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
Need to work on extending borrow scopes in some cases.

File: lib/SILOptimizer/Transforms/ConditionForwarding.cpp [hunk 1]
```diff
@@ -281,10 +280,16 @@ bool ConditionForwarding::tryOptimize(SwitchEnumInst *SEI) {
     if (HasEnumArg) {
       // The successor block has a new argument (which we created above) where
       // we have to pass the Enum.
+      assert(!getFunction()->hasOwnership() ||
+             EI->getType().isTrivial(*getFunction()));
       BranchArgs.push_back(EI);
     }
     B.createBranch(BI->getLoc(), SEDest, BranchArgs);
     BI->eraseFromParent();
+    if (EI->use_empty()) {
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply lib/SILOptimizer/Transforms/ConditionForwarding.cpp:302]: Yes, thanks for catching this! I was able to write a test with `debug_value` of `EI` in `simple_forwarding3`

**Comment by meg-gupta** [comment]
@swift-ci test

**Review by eeckstein — APPROVED** [review]
lgtm!

**Pull request merged**